editor/index.js

```diff
@@ -82,6 +86,15 @@ export function createEditorInstance( id, post ) {
 
 	preparePostState( store, post );
 
+	const warnDirtyBeforeUnload = ( event ) => {
```

Comment:
I'm not sure if this makes more sense here or further down the tree in a component like the one we have for `SavedState`.